data-serving/scripts/data-pipeline/run_pipeline.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# Executes a pipeline to fetch the latest line-list data, convert it into the
+# MongoDB schema format, and import it into the MongoDD database.
+
+readonly USAGE="Usage: $0 [-m <mongodb_connection_string>] [-d <database>] [-c <collection>] [-r <sample_rate>] [-s <schema_path>]"
+readonly SCRIPT_PATH="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly NCOV2019_REPO_PATH='/tmp/nCoV2019'
+readonly CONVERTED_DATA_PATH='cases.json'
+
+mongodb_connection_string='mongodb://127.0.0.1:27017'
+db='covid19'
+collection='cases'
+sample_rate=1
+schema_path="$SCRIPT_PATH/../../data-service/schemas/cases.schema.json"
+
+while getopts :m:d:c:r:s flag
+do
+    case "${flag}" in
+        m) mongodb_connection_string=${OPTARG};;
+        d) db=${OPTARG};;
+        c) collection=${OPTARG};;
+        r) sample_rate=${OPTARG};;
+        s) schema_path=${OPTARG};;
+        ?) echo $USAGE; exit 1
+    esac
+done
+
+function print() {
+    echo -e "\n---------------\n$1\n---------------\n"
+}
+
+function fetch_latest_data() {
+    # Check out the repo with the latest data archive and the geocoding script.
+    if [ -d $NCOV2019_REPO_PATH ]; then 
+        print "CoV2019 repo already exists at $NCOV2019_REPO_PATH; updating"
+        (cd $NCOV2019_REPO_PATH;
+         git checkout master &> /dev/null;
+         git fetch origin;
+         git reset --hard origin/master)
+    else
+        print "Cloning nCoV2019 repo to $NCOV2019_REPO_PATH"
+        git clone https://github.com/beoutbreakprepared/nCoV2019.git \
+            -- $NCOV2019_REPO_PATH
+    fi
+}
+
+function convert_data() {
+    print 'Converting the latest data archive to the MongoDB schema'
+    python3 $SCRIPT_PATH/../convert-data/convert_data.py \
```

Comment:
If you want to get fancy, you can create an appropriate venv and `source /path/to/activate` prior to this call. Probably not worth, but just in case you haven't had enough fun with this bash script yet...

Reply:
meh venv, get fancier and just use docker instead :) you get a clean env for sure this way and we all have it installed ;)

Reply:
I thought about it but I couldn't tell if this was a good use case for docker, since I'm not running any services. wdyt? (not saying I'm going to change it now but if it's worth it, down the road!)

Reply:
We all have python3 installed on our machines so it's not a big deal but being able to do `docker run kellyrepo/datapipeline` without having to install anything else would be nice for bigger projects.